TLM/TLM/UI/AllowDespawn/AllowDespawnPanel.cs

```diff
@@ -0,0 +1,256 @@
+            };
+        */
+
+        public static CheckboxOption AllowDespawnPassengerTrains =
+            new ("AllowDespawnPassengerTrain", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Passenger Trains",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.PassengerTrain, newValue),
+                Value = IsDespawnAllowed(ExtVehicleType.PassengerTrain)
+            };
+
+        /* TODO #1368 - not supported yet
+         public static CheckboxOption AllowDespawnCargoPlanes =
+            new ("AllowDespawnCargoPlane", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Cargo Planes",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.CargoPlane, newValue),
+                Value = IsDespawnAllowed(ExtVehicleType.CargoPlane)
+            };
+        */
+
+        public static CheckboxOption AllowDespawnCargoTrains =
+            new ("AllowDespawnCargoTrain", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Cargo Trains",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.CargoTrain, newValue),
+                Value = IsDespawnAllowed(ExtVehicleType.CargoTrain)
+            };
+
+        public static CheckboxOption AllowDespawnCargoTrucks =
+            new ("AllowDespawnPassengerTruck", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Cargo Trucks",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.CargoTruck, newValue),
+                Value = IsDespawnAllowed(ExtVehicleType.CargoTruck)
+            };
+
+        public static CheckboxOption AllowDespawnService =
+            new ("AllowDespawnService", Options.PersistTo.Global) {
+                Label = "AllowDespawn.Checkbox:Service vehicles",
+                Handler = (bool newValue) => OnChange(ExtVehicleType.Service, newValue),
+                Value = IsDespawnAllowed(ExtVehicleType.Service)
+            };
+
+        private static readonly Dictionary<CheckboxOption, ExtVehicleType> AllowDespawns = new ()
+        {
+            { AllowDespawnPassengerCars, ExtVehicleType.PassengerCar },
+            { AllowDespawnBuses, ExtVehicleType.Bus },
+            { AllowDespawnTaxis, ExtVehicleType.Taxi },
+            { AllowDespawnTrams, ExtVehicleType.Tram },
+            { AllowDespawnTrolleybuses, ExtVehicleType.Trolleybus },
+            /*TODO #1368 - not supported yet
+            { AllowDespawnPassengerPlanes, ExtVehicleType.PassengerPlane },
+            */
+            { AllowDespawnPassengerTrains, ExtVehicleType.PassengerTrain },
+            /*TODO #1368 - not supported yet
+             { AllowDespawnCargoPlanes, ExtVehicleType.CargoPlane },
+             */
+            { AllowDespawnCargoTrains, ExtVehicleType.CargoTrain },
+            { AllowDespawnCargoTrucks, ExtVehicleType.CargoTruck },
+            { AllowDespawnService, ExtVehicleType.Service },
+        };
+
```

Comment:
bicycles don't despawn? how do you know what can despawn and what can't?

Reply:
First you need to "block despawn" to use allow exclusion list 😉

Reply:
> First you need to "block despawn" to use allow exclusion list

how? 

Also when does Di-spawn filter apply? when the "Disable despwaing" checkbox is On or Off? I think you should grey out the UI for the filter when it does not apply.

Reply:
Only when "disable despawning" is on (I'll disable that button when that feature is not active).
It's exclusion list of "all", but it was never really "all" vehicles. 
That "all" applies _**only to road and rail vehicle types**_ we manage, so anything like bikes, ships, airplanes, helicopters etc. are ignored completely because we don't patch the code that could block despawning.

Reply:
There's a `ReadOnly` property on the `ActionButton` UI wrapper which should work (if not lmk and I will fix it).